### DISCUSSION TITLE
uncaught exception

### DISCUSSION THREAD

**Discussion by vuywoi** [Monika-After-Story/MonikaModDev#9303] [QA - tech support + general] [answered]
i tried adding spritepacks into the mod and i dont know if i did anything wrong but this happened

I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/definitions.rpy", line 516, in <module>
  File "game/definitions.rpy", line 1639, in Event
AttributeError: 'StoreModule' object has no attribute 'mas_utils'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\bootstrap.py", line 295, in bootstrap
    renpy.main.main()
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\main.py", line 364, in main
    renpy.game.script.load_script()  # sets renpy.game.script.
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\script.py", line 265, in load_script
    self.load_appropriate_file(".rpyc", ".rpy", dir, fn, initcode)
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\script.py", line 753, in load_appropriate_file
    self.finish_load(stmts, initcode, filename=fn + source)
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\script.py", line 445, in finish_load
    node.early_execute()
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\ast.py", line 865, in early_execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\Users\dalen\Downloads\Doki Doki Literature Club\renpy\python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/definitions.rpy", line 516, in <module>
  File "game/definitions.rpy", line 1639, in Event
AttributeError: 'StoreModule' object has no attribute 'mas_utils'

Windows-8-6.2.9200
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.8.6-unstable.2022.06.22.161

**Comment by Booplicate** [comment, marked as answer]
Looks like you deleted some scripts, you'll have to reinstall the mod.

**Reply by vuywoi** [reply]
will it reset my progress? if so, how can i continue it?

**Reply by Booplicate** [reply]
No, you won't lose your progress (maybe some spritepacks/submods).

**Reply by vuywoi** [reply]
how can i reinstall the mod? do i delete everything from the previous then install again? i already installed it again, replacing the old ones (i have a persistent just incase) and it still had the uncaught exception. im very sorry im kinda slow

**Reply by Booplicate** [reply]
Since you're using unstable, it'll be more hassle for you. You need to [download](https://ddlc.moe/) and install DDLC, then [download](https://monikaafterstory.com/releases.html) and install MAS, run it and update, then try to launch again and it should work.

**Reply by vuywoi** [reply]
i remember it saying that it was unable to use unstable when i updated it so i had to use repeat. also wont i have to do the whole story again if i reinstall ddlc (i used steam to download ddlc)

**Reply by Booplicate** [reply]
If you get a new traceback, send it here.

**Reply by vuywoi** [reply]
so, i reinstalled ddlc and the mod but it restarted my progress in the mod.. is there anything i can do to fix it?

**Reply by Booplicate** [reply]
Did you delete your persistent?

**Reply by vuywoi** [reply]
no, i still have it. is there a way to use my persistent?

**Reply by Booplicate** [reply]
Did it suggest to update MAS when you launched it?

**Reply by vuywoi** [reply]
i downloaded it from the latest releases, before and after.

**Reply by Booplicate** [reply]
Yes, but did it suggest to update MAS when you launched it? Can you screenshot our persistents?

**Reply by vuywoi** [reply]
it didn't suggest it. 
![Screenshot 2022-07-13 220509](https://user-images.githubusercontent.com/109251327/178882334-d0061154-2eb6-4e8a-86fa-ad8c06a65e45.png)

**Reply by Booplicate** [reply]
Did you see Chibi dialogue with black screen on startup at all?

**Reply by vuywoi** [reply]
yes, it said to delete the "persistent unstable" file, why?

**Reply by Booplicate** [reply]
It also suggested to update the mod, you should've clicked that. Not deleting your persistent.

**Reply by Booplicate** [reply]
Backups all your persistents to another folder (or better a cloud), update the mod to unstable version, restore one of persistents backups.

**Reply by vuywoi** [reply]
it worked! tysm, i really appreciate your help!
